lib/shared/errors.js

```diff
@@ -228,7 +234,7 @@ exports.getDescription = (error) => {
   }
 
   if (error.stack) {
```

Comment:
If we let it carry on, then the error object will get stringified, which is the worst possible case. I do get your point though, the `considerStacktrace` option is confusing if it also means an empty string is returned right away if there is no stack trace.

One solution would be to remove the string transformation fallback at the bottom of this function, however that would be really useful when dealing with strange errors coming from third party libraries that don't adhere to the NodeJS "guidelines" (e.g: a library throwing an object that is not a common `Error` instance).

Another solution would be to rephrase the option name to signify that we want a user friendly description, or nothing at all (thus ignoring non-user, but developer friendly things like stack traces and stringified versions of object). We could call this option `ignoreNonUserFriendlyForms()` or something like that.

What do you think?